pkg/etcd/s3.go

```diff
@@ -334,7 +334,12 @@ func (s *S3) deleteSnapshot(ctx context.Context, key string) error {
 	ctx, cancel := context.WithTimeout(ctx, s.config.EtcdS3Timeout)
 	defer cancel()
 
-	key = path.Join(s.config.EtcdS3Folder, key)
+	// verifys if the folder was added before the key to delete the snapshot
+	// since the retention can't delete the snapshot if has the s3 folder pre defined
+	if !strings.Contains(key, s.config.EtcdS3Folder+"/") {
```

Comment:
```suggestion
	if !strings.HasPrefix(key, s.config.EtcdS3Folder) {
```

Reply:
It sounds like the problem here is that deleteSnapshot is being passed the snapshot path (folder+name) instead of just the name, as expected. Does this perhaps need to be fixed in whatever's calling deleteSnapshot instead?

Reply:
You might also add a test for this at https://github.com/k3s-io/k3s/blob/master/tests/e2e/s3/s3_test.go#L129-L130 - right now we only test manual prune, apparently there are no tests for scheduled snapshot retention.

Reply:
https://github.com/k3s-io/k3s/blob/df5db28a687f5fb6fd0f5cb68873d2184974e953/pkg/etcd/s3.go#L337

In this case this is the part that has the bug, because when we receive the key parameter with folder+key we add the folder again to the key, resulting in folder+folder+key

Reply:
When using S3 it is expected that you just pass in the snapshot name when interacting with snapshots. That is for example how snapshots are restored or deleted.

It sounds like the scheduled snapshot retention is passing the full path into this function, instead of the basename, instead when pruning. You might investigate where that is occurring and fix it there, instead of adding a work-around here.